xcode/Subconscious/Shared/Components/Detail/DetailStackView.swift

```diff
@@ -288,61 +287,85 @@ struct DetailStackModel: Hashable, ModelProtocol {
         return Update(state: model)
     }
     
-    static func rebaseLinkOnAddress(
-        address: Slashlink,
-        link: SubSlashlinkLink
-    ) -> Slashlink {
-        // Stitch the base address on to the tapped link, making any
-        // bare slashlinks relative to the sphere they belong to.
-        //
-        // This is needed in the viewer but address will always based
-        // on our sphere in the editor case.
-        guard case let .petname(basePetname) = address.peer else {
-            return link.slashlink
+    public static func findBestAddressForLink(
+        context: Peer?,
+        link: SubSlashlinkLink,
+        environment: TestableEnvironment
```

Comment:
All the logic for mapping a link to a navigable address is now contained in this function. There are tests for this, but not to full coverage since traversal is impossible in testing atm.

Did my best to document the thinking, please flag any confusion or tweaks to make.

Reply:
This looks good. A few questions:

- I see 3x calls to Noosphere. Is this more than we were making before, or the same?
- AddressBookService uses Noosphere under the hood. Could this method could become an extension of NoosphereService or Sphere instead of an environment function?